### ISSUE TITLE
502 It looks like the dev server has not been started yet. Start from the editor ↗

### ISSUE THREAD

**Issue by huythai1602** [codesandbox/codesandbox-client#8442]
I created a devbox in codesandbox and my preview really said this:
502
It looks like the dev server has not been started yet. [Start from the editor ↗]
https://codesandbox.io/p/sandbox/photo-sharing-mrdmvk

**Comment by filipelima18** [comment]
Hi. We released a fix today that solved some issues related to VM starts. Opening that URL just now, I was able to load it with no issues on my end. Can you check?

**Comment by CompuIves** [comment]
Can you share the link @dushmanta05? This message is given when the dev server is not started on the opened port, opening the editor should fix it, or starting a dev server on that port fixes it.

**Comment by dushmanta05** [comment]
@CompuIves Unfortunately, I can't share because of confidentiality because the project is my work. I was trying CodeSandbox because I faced some issues in my local. I'll try to provide as much information as I can tomorrow. Thanks.

**Comment by CompuIves** [comment]
Okay, well, in any case it should work if you start the dev server.

**Comment by parnika30** [comment]
how to resolve this issue?

**Comment by jamalbernhard** [comment]
I am seeing this issue in ALL of my devboxes right now and it's extremely frustrating. I'm supposed to be giving a coding interview in 10 hours and it looks like I will have to cancel and reschedule because I can't get preview windows working. When I click "Start from the editor" it just seems to load a new editor instance inside the preview (right-hand) pane, and continuing to click "Open preview" after that does nothing new. Grrrrr.

**Comment by cfenzo** [comment]
I ran into the same issue after upgrading a devbox to the latest version,  using the "React + Tailwind" template.
It tries to preview "dev:5173", and I get the error: This task of this preview has a port configured, but it's not opened.
<img width="760" alt="Screenshot 2024-06-20 at 07 18 03" src="https://github.com/codesandbox/codesandbox-client/assets/59037/411a5b35-725d-48a3-8d85-b1dcb663dfab">

If I try to open the correct csb.app url in a tab, the error message is the same as the one in the title here.
<img width="388" alt="Screenshot 2024-06-20 at 07 18 09" src="https://github.com/codesandbox/codesandbox-client/assets/59037/491a5dc3-90ad-4595-adb5-896b0c189d70">

I can see in the terminal for the current task (dev), that the vite command has no errors and is listening on port 5173.
<img width="436" alt="Screenshot 2024-06-20 at 07 17 54" src="https://github.com/codesandbox/codesandbox-client/assets/59037/230c30df-11d4-4773-abd0-2297dfaec0ab">

In console there's an error:
<img width="714" alt="Screenshot 2024-06-20 at 07 38 48" src="https://github.com/codesandbox/codesandbox-client/assets/59037/f008fff7-86f2-4c8f-84f5-1fbd7464eb0e">

I tried to restart the VM and also reload the window a couple more times, and now the preview tab is gone, and there is no choice in any menu for "open preview", only "open documentation".
<img width="1792" alt="Screenshot 2024-06-20 at 07 43 51" src="https://github.com/codesandbox/codesandbox-client/assets/59037/f139e77d-0d48-4af2-ac4c-adb1e8658db5">

 I've cleared the application data in dev tools and tried the "Reset choice for 'file operation needs preview'" that I've previously have had to use to get the preview back, but nothing changes.


I made a fork of the devbox and removed all  client code. It has the same issues:  https://codesandbox.io/p/devbox/failing-devbox-forked-j4v5v8

**Comment by jamalbernhard** [comment]
@cfenzo  Yes updating to the latest VM seems to be causing the issue. Even if I just start a brand new React + TS devbox from their Vite template and then immediately update to the latest VM, preview is broken!

**Comment by jarvis8x7b** [comment]
using the API to create devboxes and facing this issue as well, previously devboxes always started by themselves and preview was available but now it's always "502 It looks like the dev server has not been started yet". is there a way to programmatically start the devboxes?

**Comment by vivekanand04** [comment]
502
It looks like the dev server has not been started yet. [Start from the editor ↗](https://codesandbox.io/p/redirect-to-project-editor/jvkzf7)      why is happening ????????????

**Comment by vivekanand04** [comment]
solution
i have same problem but have a backup idea 
make a new file/folder in codesandbox and copy paste your this code into new created folder.

**Comment by ifakhfakh** [comment]
I found solution for this. Try opening the terminal then run: ng serve . If it tells you that ng command is not known, then execute 
sudo npm install -g @angular/cli. After that run ng serve. After it finishes, you will see a small pop-in with two options: open preview or open externally.

**Comment by filipelima18** [comment]
Hi. Last Friday, we released a fix for some of the most common issues that prevented previews from starting. To get the fix to work:
1. Update the editor to the latest version if needed (a notification will show up on the top left of the editor if it's outdated).
2. Restart the VM by clicking the CodeSandbox logo at the top left and then "Restart Devbox".

**Comment by ryanbas21** [comment]
I see this error on a fresh sandbox i just made, using react / webpack.

**Comment by 1chary** [comment]
Could any one help me out i also am facing the same issue opened new project and connected with git repo and showing this error in the preview 502
It looks like the dev server has not been started yet. Start from the editor when clicked on this it was redirected to the next page and again showing the same error..

**Comment by filipelima18** [comment]
Hi @1chary. In case this is a public repo, could you share its URL here so we can have a look? If it's a private repo, please share some screenshots of your editor with support@codesandbox.io so we can investigate.

**Comment by silva-091** [comment]
how to fix 502 erro dev server has not started yet . i have same issue

**Comment by CompuIves** [comment]
The root cause of this issue is that the dev server inside the VM is not started. You can open a terminal inside the VM to start it. Ultimately, the cause for this can be that the dev server crashed, or that the dev server ran out of memory. This is something that is not caused by CodeSandbox, but is similar to if a dev server on your own computer stopped running and you need to start it again.

There is one instance where a dev server has opened a port, but we give back a 502. I _think_ that can be caused by the fact that the dev server opened the port on `localhost`, while it should open the port on `0.0.0.0`. You can often configure the dev server to do this by passing `--host 0.0.0.0` to the command that starts the dev server.

If you are experiencing this error, link a devbox and I'll take a look!

**Comment by reekotubbs** [comment]
I am receiving the same error. Please check the following repo https://github.com/sagarshah-techno/OutlookRulesReader
